lib/utils/httputil/httputil.go

```diff
@@ -384,13 +393,37 @@ func newRequest(method string, opts sendOptions) (*http.Request, error) {
 	return req, nil
 }
 
+func fallbackToHTTP(
+	client *http.Client, method string, opts *sendOptions) (*http.Response, error) {
+
+	req, err := newRequest(method, opts)
+	if err != nil {
+		return nil, err
+	}
+	req.URL.Scheme = "http"
+
+	return client.Do(req)
+}
+
+func shouldFallbackToHTTP(req *http.Request, resp *http.Response, opts *sendOptions) bool {
+	if req.URL.Scheme == "http" { // Already in HTTP.
+		return false
+	}
+	// Try fallback on non-retryable errors.
```

Comment:
I'm fine with this as it matches the previous logic. I'm wondering however why we don't know the specific HTTP status code returned in case we need to do a HTTP fallback. It seems like we should get the same status code always in that case.

Reply:
When client sent an https request to http server, client.Do returns a "network error" but the error is not typed and resp can be nil...so there is no specific indicator of when we would need to do a HTTP fallback.